kernel/include/ksched.h

```diff
@@ -239,13 +239,15 @@ static inline bool _is_valid_prio(int prio, void *entry_point)
 	return true;
 }
 
-static inline void _ready_one_thread(_wait_q_t *wq)
+static inline struct k_thread *_ready_one_thread(_wait_q_t *wq)
```

Comment:
No objections to this API change, especially for an inline function where it incurs no overhead.

But I'm a little worried about the use/abuse of the swap return value in the pthread code here.  That's a voodoo-scale interface that I'd like to get rid of, precisely because it engenders bugs like this.  Zero isn't "success", it's just whatever the subsystem wants it to mean.  The default is indeed -EAGAIN (to indicate that something "unexpected" rescheduled the thread, like a timeout).  If pthread_cond_wait() was surprised here, it's probably going to be surprised elsewhere too.